hashes/src/lib.rs

```diff
@@ -102,8 +102,7 @@
 #[cfg(bench)] extern crate test;
 #[cfg(any(test, feature = "std"))] extern crate core;
 #[cfg(feature = "core2")] extern crate core2;
-#[cfg(feature = "alloc")] extern crate alloc;
-#[cfg(all(not(feature = "alloc"), feature = "std"))] use std as alloc;
+#[cfg(all(feature = "alloc", not(feature = "std")))] extern crate alloc;
```

Comment:
FTR, it's perfectly OK to import the crate unconditionally.

Reply:
That won't break no-std builds that do not want to use an allocator?

Reply:
Oh, crap, you're right. The condition could be just `cfg(feature = "alloc")` though.